ros2_socketcan_msgs/msg/FdFrame.msg

```diff
@@ -1,8 +1,6 @@
 std_msgs/Header header
 uint32 id
-bool is_rtr
 bool is_extended
 bool is_error
-bool is_fd_frame
```

Comment:
Hello, 

On ROS1 I am using the same msg for both CAN and CANFD frame (https://github.com/xbroquer/ros_canopen/blob/canfd/noetic-devel/can_msgs/msg/Frame.msg)  dlc field is filled with len.

I am thinking to switch to this new ros2_socketcan message.

Is it possible to keep it also compatible with classic CAN (keep rtr flag , dlc, is_fd_frame) and  add the len ?

Reply:
@xbroquer Unfortunately, this is not realistic for several reasons:

1. `can_msgs/Frame` is the de-facto standard for both ROS1 and ROS2 for "traditional" CAN.
2. `ros2_socketcan` has been using `can_msgs/Frame` since ROS2 Dashing, meaning there are existing recordings, codebases, etc. that assume that this is the message type that will be produced by default from this package.
3. My company (and possibly others) need the ability to use CAN FD with older ROS2 versions like Foxy and breaking the message API on this old of a version is not reasonable so it must produce different messages for traditional and FD.

Reply:
My proposal was not to remove the default can_msgs/Frame. 

Instead of having the enable_fd parameter to switch from default ( can HS with can_msgs/Frame) to canfd (with ros2_socketcan::Frame) , I propose to use a parameter to select the output interface ( like enable_fd_interface).

When "enable_fd_interface" will be activated, we could use the same ros2_socketcan::Frame for both can and canfd (I am ok keeping dlc for len)

In my ROS1 implementation I use the MTU property to determine the can type:
https://github.com/xbroquer/ros_canopen/blob/3dba73217b8c7af94d59f2130b08a5661d0a909e/socketcan_interface/include/socketcan_interface/socketcan.h#L205